### PR TITLE
Make page performance overlay work with flattened reports

### LINKED ISSUE #17807: Page Performance reports don't show data (empty graph) when viewing performance for a flattened page URL
Page Performance reports work well in the default "Pages" report's hierarchical view,
but as soon as the report is flattened (a very often used feature),
then Page performance reports don't work (empty graph).

## Expected Behavior

Graph with data eg:

Screenshot from 2021-07-23 16-19-50

## Current Behavior

Screenshot from 2021-07-23 16-17-34

## Steps to Reproduce (for Bugs)
* Go to Behaviour > Pages report.
* Flatten the report
* Hover on a row and click "open page performance report"
* The report displays no data

Note: on the demo it does not anyway as it's missing performance data. But can be reproduce in any other matomo

## Your Environment

* Matomo Version: 4.3.x

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

Pages reports provide a row action that will open the page performance report for a specific row. When viewing any action report in normal view that works like a charm. But when the report is changed to be flattened the overlay doesn't show any data anymore.

The reason for this is, that the API request performed for the overlay didn't get the `flat` parameter passed through. This causes the provided label (action url) not to be found in the report.

Passing through the `flat`  parameter fixes this problem and the data is displayed correctly again.

fixes #17807 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/PagePerformance/javascripts/rowaction.js [hunk 1]
```diff
@@ -33,20 +33,21 @@
...
     };
 
     DataTable_RowActions_PagePerformance.prototype.doOpenPopover = function (urlParam) {
         var urlParamParts = urlParam.split(':');
         var apiMethod = urlParamParts.shift();
         var label = decodeURIComponent(urlParamParts.shift());
+        var isReportFlat = urlParamParts.shift();
 
-        PagePerformance.show(apiMethod, label);
+        PagePerformance.show(apiMethod, label, isReportFlat);
```
Comment (caddoo) [inline plugins/PagePerformance/javascripts/rowaction.js:51]: this parameter name confuses me `isReportFlat` 

I would expect a boolean but it seems you are passing an item from `urlParamParts` (whatever that is) or `undefined`.

Would casting it make it a bit clearer ?

Comment (michalkleiner) [reply plugins/PagePerformance/javascripts/rowaction.js:51]: If I understand the change correctly I think it's a 0/1 int-encoded boolean as you can't easily type boolean in URLs.

File: plugins/PagePerformance/javascripts/rowaction.js [hunk 2]
```diff
@@ -33,20 +33,21 @@
 
     DataTable_RowActions_PagePerformance.prototype.performAction = function (label, tr, e, originalRow) {
         var apiMethod = this.dataTable.param.module + '.' + this.dataTable.param.action;
-        this.openPopover(apiMethod, label);
+        this.openPopover(apiMethod, label, this.dataTable.param.flat || 0);
```
Comment (michalkleiner) [inline plugins/PagePerformance/javascripts/rowaction.js]: not a functional change but looks a bit more consistent with the rest of the change
```suggestion
        var isReportFlat = this.dataTable.param.flat || 0;
        this.openPopover(apiMethod, label, isReportFlat);
```

File: plugins/PagePerformance/javascripts/rowaction.js (hunk 1 above)
Comment (sgiehl) [reply plugins/PagePerformance/javascripts/rowaction.js:51]: exactly

**Review by caddoo — APPROVED** [review]

**Pull request merged**